plugins/edsm.py

```diff
@@ -331,87 +471,116 @@ def cmdr_data(data, is_beta):
 
                 if should_send(pending):
-                    (username, apikey) = credentials(cmdr)
+                    creds = credentials(cmdr)  # TODO: possibly unbound
+                    if creds is None:
+                        raise ValueError("Unexpected lack of credentials")
+
+                    (username, apikey) = creds
                     data = {
                         'commanderName': username.encode('utf-8'),
                         'apiKey': apikey,
                         'fromSoftware': applongname,
                         'fromSoftwareVersion': appversion,
                         'message': json.dumps(pending, ensure_ascii=False).encode('utf-8'),
                     }
+
                     r = this.session.post('https://www.edsm.net/api-journal-v1', data=data, timeout=_TIMEOUT)
                     r.raise_for_status()
                     reply = r.json()
-                    (msgnum, msg) = reply['msgnum'], reply['msg']
+                    (msg_num, msg) = reply['msgnum'], reply['msg']
                     # 1xx = OK
                     # 2xx = fatal error
                     # 3&4xx not generated at top-level
                     # 5xx = error but events saved for later processing
-                    if msgnum // 100 == 2:
-                        logger.warning(f'EDSM\t{msgnum} {msg}\t{json.dumps(pending, separators = (",", ": "))}')
+                    if msg_num // 100 == 2:
+                        logger.warning(f'EDSM\t{msg_num} {msg}\t{json.dumps(pending, separators=(",", ": "))}')
                         plug.show_error(_('Error: EDSM {MSG}').format(MSG=msg))
+
                     else:
                         for e, r in zip(pending, reply['events']):
-                            if not closing and e['event'] in ['StartUp', 'Location', 'FSDJump', 'CarrierJump']:
+                            if not closing and e['event'] in ('StartUp', 'Location', 'FSDJump', 'CarrierJump'):
                                 # Update main window's system status
                                 this.lastlookup = r
                                 # calls update_status in main thread
                                 this.system_link.event_generate('<<EDSMStatus>>', when="tail")
+
                             elif r['msgnum'] // 100 != 1:
                                 logger.warning(f'EDSM\t{r["msgnum"]} {r["msg"]}\t'
                                                f'{json.dumps(e, separators = (",", ": "))}')
+
                         pending = []
 
                 break
             except Exception as e:
                 logger.debug('Sending API events', exc_info=e)
                 retrying += 1
+
         else:
             plug.show_error(_("Error: Can't connect to EDSM"))
 
         if closing:
             return
 
 
-# Whether any of the entries should be sent immediately
-def should_send(entries):
+def should_send(entries: List[Mapping[str, Any]]) -> bool:
+    """
```

Comment:
The navbeaconscan is about sending immediately, but the rest of the code is whether the event is sent at all.